examples/LoggingExtensions/Program.cs

```diff
@@ -30,9 +29,18 @@
         .AddConsoleExporter();
 });
 
+// Creates an OpenTelemetryEventSourceLogEmitter for routing EventSources with
+// names matching OpenTelemetry* into logs
+using var openTelemetryEventSourceLogEmitter = new OpenTelemetryEventSourceLogEmitter(
+    openTelemetryLoggerProvider, // <- Events will be written to openTelemetryLoggerProvider
+    (name) => name.StartsWith("OpenTelemetry") ? EventLevel.LogAlways : null,
```

Comment:
Was it intentional that "OpenTelemetry" was used? any risk of cyclic/loops? 
Might be good to create a custom EventSource-Foo or something, and use that in examples?

Reply:
Ya it was intentional. I just added this comment to explain the behavior:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/83974706f2510f1a6885eaad1e8294c2e487954c/examples/LoggingExtensions/Program.cs#L56-L58

I'm open to changing it LMK what you think about that.

Doesn't have any cyclic or loop issues currently as OpenTelemetry.Extensions.EventSource doesn't have its own EventSource.

Reply:
I see.
I meant this when I said cyclic loop:

If I enable this EventSourceLogEmitter, it'll subscribe to all ES starting with OpenTelemetry, and pipes them to OTel Logging SDK.

OTel Logging SDK internally uses ES whose name starts with OpenTelemetry. So that EventSource Logs gets piped through this LoggingSDK again, which inturn produce more log....

(or did i got it wrong fully 🏃 )

Reply:
The enduser for this package is people who already EventSource for their logging, and want those piped through OTel logging. For such use-cases, the best would be to show a custom EventSource, logging to it, but it showing up in OTel logging.
Similar to the example for Serilog...

Reply:
* Updated the example to use a custom source.

* For circular logging, I think we may have a bigger issue. Opened #3470 to discuss.